API.md

```diff
@@ -62,6 +62,11 @@ If no argument is provided, the method reads all channels. Otherwise, the data p
 
 If the current sensor reading does not allow understanding of the line position (some trash under the sensor or gone out of track), the previous successful estimation (which is cached) is returned. The starting value is `0.0`.
 
+### `uint8_t autoCalibrate(void)`
+### `uint8_t autoCalibrate(enum CALIBRATE_MODE mode)`
+
+Returns optimal sensitivity value for current octoliner/track pair. Mode can be: `CALIBRATE_MODE::DRAFT` (quick but can be not accuracy: about 2 sec), `CALIBRATE_MODE::NORMAL` (optimal: about 8 sec), `CALIBRATE_MODE::ACCURACY` (for a long time: about 38 sec). Without parameters autoCalibrate use parameter NORMAL.
+
```

Comment:
That’s why we should always start with documentation…

First, the terminology doesn’t play well: `setSensitivity` vs `autoCalibrate`. We calibrate sensitivity, thus a better name would be `calibrateSensitivity` or `optimizeSensitivity`. I find the latter slightly more clear.

Then, I see no reason to complicate things and introduce an extra level of indirection: the `enum CALIBRATE_MODE`. It means the single thing, the settling time of a single measurement. We could make it mean exactly this, don’t we? Useful presets could be defined as simple constants then:

```cpp
constexpr uint32_t OCTOLINER_ACCURACY_DRAFT = 120;
constexpr uint32_t OCTOLINER_ACCURACY_NORMAL = 480;
constexpr uint32_t OCTOLINER_ACCURACY_BEST = 2400;
```

Next, how the function behaves when it fails? For example, when the sensor is exposed to the void. This should be clearly defined. Now I see it will have “Integer divided by zero” error, not good. What should happen if the calibration fails after all? I think it’s up to the user to decide.

Putting all together, I suggest the following:

```cpp
### bool optimizeSensitivity();
### bool optimizeSensitivity(uint32_t measurementTimeMs);

Performs automatic sensitivity set up by making six subsequent measurements at different sensitivity levels.
The sensitivity level at which the contrast of the image ahead of Octoliner is the best wins. Make sure to put
the sensor over a contrast image like a line drawn on the surface before calling this method. For example, call
`optimizeSensitivity` in your `setup` and reset your robot to get the optimal sensitivity for the ride session.

Each measurement iteration has to settle down to make its result adequate. The time in milliseconds per
iteration is given through `measurementTimeMs` parameter. The following constants are defined as sane
choices:

- `OCTOLINER_ACCURACY_DRAFT` = 120 ms (2 sec total)
- `OCTOLINER_ACCURACY_NORMAL` = 480 ms (8 sec total, default parameter is omitted)
- `OCTOLINER_ACCURACY_BEST` = 2400 ms (38 sec total)

The method returns `true` if calibrated successfully. Use `getSensitivity` if you would like to know the value set.
If optimization fails, for example, when the sensor is exposed to void, the method returns `false`. In the case of failure, the currently set sensitivity level is left intact.

### uint8_t getSensitivity() const;

Returns the current sensitivity level previously set with `setSensitivity` or `optimizeSensitivity`.
```

Reply:
Yes, it is better then my version. I will rewrite code. But...where is 'Now I see it will have “Integer divided by zero” error'? Don`t look it.

Reply:
Oops, missed the line:

```cpp
downCnt = (downCnt == 0) ? 1 : downCnt;
```

Indeed, it will never divide by 0;

Reply:
This code removed